@here/olp-sdk-dataservice-read/lib/utils/RequestBuilderFactory.ts

```diff
@@ -27,12 +27,14 @@ import {
     OlpClientSettings
 } from "..";
 
+const MILLISECONDS_IN_SECOND = 1000;
 /**
  * A helper utils that makes the `Request` object with the base URLs of the API Lookup Service, token callback, and download manager.
  *
  * Also, you can use it to get the base URLs of the API Lookup Service.
  */
 export class RequestFactory {
+    private static refreshApiURLsTime = new Date().getTime();
```

Comment:
Can we use `refreshApiURLsTime` for each request, not one for all? Maybe we can create something like `ApiCacheRepository` for caching pairs: request-key -> expireTime?

Reply:
Changed. Now we save max-age for each URL in our `ApiCacheRepository`